src/python/bot/minimizer/js_minimizer.py

```diff
@@ -22,6 +22,13 @@
 from . import utils
 
 
+def step_back_while(cur_index, condition):
+  """Helper function. Decreases index from cur until condition is satisfied."""
+  while cur_index >= 0 and condition(cur_index):
+    cur_index -= 1
+  return cur_index
```

Comment:
nit: this may return `-1`, will the calling code handle it?

Reply:
Since python allows negative indexing, it will not error out, but rather form an incorrect hypothesis that includes the last token of the file. This is OK because if the while loop doesn't meet the condition before the beginning of the file, then the hypothesis it was trying to form was incorrect anyway.